crucible-mir/src/Mir/TransCustom.hs

```diff
@@ -556,24 +558,37 @@ ptr_slice_from_raw_parts_mut =
     ( ["core", "ptr", "slice_from_raw_parts_mut", "crucible_slice_from_raw_parts_hook"]
     , ptr_slice_from_raw_parts_impl)
 
+-- | [@@std::ptr::read](https://doc.rust-lang.org/std/ptr/fn.read.html)
+ptr_read = (["core", "ptr", "read"], ptr_read_impl "ptr::write")
 
-ptr_read :: (ExplodedDefId, CustomRHS)
-ptr_read = ( ["core", "ptr", "read"], \substs -> case substs of
+-- | [@@std::ptr::write](https://doc.rust-lang.org/std/ptr/fn.write.html)
+ptr_write = (["core", "ptr", "write"], ptr_write_impl "ptr::write")
+
+-- | [@std::ptr::read_volatile@](https://doc.rust-lang.org/std/ptr/fn.read_volatile.html)
+volatile_load = (["core", "intrinsics", "volatile_load"], ptr_read_impl "intrinsics::volatile_load")
+
+-- | [@std::ptr::write_volatile@](https://doc.rust-lang.org/std/ptr/fn.write_volatile.html)
+volatile_write = (["core", "intrinsics", "volatile_store"], ptr_write_impl "intrinsics::volatile_write")
```

Comment:
The code sharing here is good, but please do keep the top-level `:: (ExplodedDefId, CustomRHS)` type signatures for each of these four definitions. (If nothing else, omitting the type signatures will cause GHC to warn.)